dask_expr/_merge.py

```diff
@@ -99,6 +99,20 @@ def _npartitions(self):
             return self.operand("_npartitions")
         return max(self.left.npartitions, self.right.npartitions)
 
+    @property
+    def _bcast_left(self):
+        if self.operand("_npartitions") is not None:
+            if self.broadcast_side == "right":
+                return Repartition(self.left, new_partitions=self._npartitions)
+        return self.left
+
+    @property
+    def _bcast_right(self):
```

Comment:
nit: I find the name here mildly confusing, `_bcast_left` suggests to me that it's a left-sided broadcast, but it's the left side of a broadcast where _right_ is being broadcast. I don't really have a better suggestion for now though. This is non-blocking.